packages/playground/blueprints/src/lib/steps/install-asset.ts

```diff
@@ -35,15 +35,23 @@ export async function installAsset(
 		targetPath,
 		zipFile,
 		ifAlreadyInstalled = 'overwrite',
-		targetFolderName = ''
+		targetFolderName = '',
 	}: InstallAssetOptions
 ): Promise<{
 	assetFolderPath: string;
 	assetFolderName: string;
 }> {
 	// Extract to temporary folder so we can find asset folder name
 	const zipFileName = zipFile.name;
-	const assetNameGuess = zipFileName.replace(/\.zip$/, '');
+	let assetNameGuess = zipFileName.replace(/\.zip$/, '');
+	/**
+	 * Some resource types like GitDirectoryReference and UrlReference
+	 * don't have a name, so we need to generate a random name to ensure
+	 * the asset folder name exists.
+	 */
+	if (!zipFileName) {
+		assetNameGuess = `${randomString(32, '')}`;
```

Comment:
Technically this could break if the path already exists, but I wasn't sure if checking the path would overcomplicate the implementation.